### PR TITLE
create request

### PR DESCRIPTION
**Pull request by nghiabk**

### CODE REVIEW

File: app/controllers/requests_controller.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+class RequestsController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @requests = current_user.requests
+    @requests = @requests.paginate page: params[:page], per_page: 15
```
Comment (HungSonSN) [inline app/controllers/requests_controller.rb]: current_user.requests.paginate

File: app/controllers/admin/requests_controller.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+class Admin::RequestsController < ApplicationController
+  before_action :authenticate_user!  
+  before_action :admin_user
+  
+  def index    
+    @requests = Request.all
+    @requests = @requests.paginate page: params[:page], per_page: 15
```
Comment (HungSonSN) [inline app/controllers/admin/requests_controller.rb]: Request.all.paginate

**Comment by HungSonSN** [comment]
ok

File: app/controllers/requests_controller.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      flash[:success] = "Request is created!"
+      redirect_to requests_url    
+    else
+      render 'new'
+    end
+
+  end
+
+  def destroy
+   	Request.find(params[:id]).destroy
```
Comment (dieunb) [inline app/controllers/requests_controller.rb]: indent

File: app/models/request.rb [hunk 4]
```diff
@@ -1,5 +1,13 @@
 class Request < ActiveRecord::Base
   belongs_to :user
+  default_scope -> {order(created_at: :desc)}
```
Comment (dieunb) [inline app/models/request.rb]: remove default_scope

File: config/routes.rb [hunk 5]
```diff
@@ -7,21 +7,22 @@
...
 
   namespace :admin do
-    root 'categories#new'
+    root 'requests#index'
     resources :categories 
     resources :books do
       resources :image_books
     end
+    resources :requests, only: [:index, :update]
+    patch 'update_request' => 'requests#update'
```
Comment (dieunb) [inline config/routes.rb]: remove

**Pull request merged**